src/hotspot/share/gc/parallel/psParallelCompact.cpp

```diff
@@ -2551,7 +2551,6 @@ void PSParallelCompact::verify_complete(SpaceId space_id) {
 #endif  // #ifdef ASSERT
 
 inline void UpdateOnlyClosure::do_addr(HeapWord* addr) {
-  _start_array->update_for_block(addr, addr + cast_to_oop(addr)->size());
```

Comment:
Currently, BOT entries are verified on-update, the last statement of `update_for_block_work`. In my experience, they are rarely corrupted.

Reply:
That is my suggestion: because they are not updated any more, there will be no verification. You are right that they are rarely corrupted so that the VM crashes, but if they are not optimal, hard to detect performance issues may creep in.

Reply:
`ObjectStartArray::object_start` contains assertions about BOT being precise, just fyi.

Reply:
That's correct, but it would be better to detect this asap. A lot of time could have passed until this is discovered there.